luigi/worker.py

```diff
@@ -389,19 +436,17 @@ class TracebackWrapper:
     """
 
     def __init__(self, trace):
+        warnings.warn("{} is deprecated and will be removed in a future release".format(
+                      self.__class__.__name__), DeprecationWarning)
         self.trace = trace
 
 
-def check_complete(task, out_queue):
+def check_complete(task):
     """
-    Checks if task is complete, puts the result to out_queue.
+    Checks if task is complete.
     """
     logger.debug("Checking if %s is complete", task)
-    try:
-        is_complete = task.complete()
-    except Exception:
-        is_complete = TracebackWrapper(traceback.format_exc())
-    out_queue.put((task, is_complete))
+    return task.complete()
```

Comment:
What's the potential for hitting an exception here? Was the previous catch never caught and thus pointless? Or are we introducing the opportunity here for an unhandled exception?